src/go/metrics/collector.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+)
+
+type metricsCollector struct {
+	sender MetricSender
+	stop   chan bool
+}
+
+type MetricsCollectionOptions struct {
+	Delay time.Duration
+	Clock clock.Clock
+}
+
+var collectorOnce sync.Once
+var collectorInstance *metricsCollector
+
+func DefaultMetricsCollectionOpts() *MetricsCollectionOptions {
+	return &MetricsCollectionOptions{
+		Delay: time.Minute,
+		Clock: clock.New(),
+	}
+}
+
+func StartMetricsCollection(options *MetricsCollectionOptions) error {
```

Comment:
I would turn this into:

`func NewCollector(c clock.Clock, period time.Duration, sender Sender) *Collector`

Collector would have two funcs:

```
func (c *Collector) RegisterGauge(Gauge) { ... }
func (c *Collector) Start() { ... }
```

No more `Registry` (another global)

Lastly, `Sender` becomes:

```
type Sender interface {
  Send(*metricsd.MetricsContainer) error
}
```

`Collector` is responsible for processing all `Gauge` samples every `period` and calling `Sender`